lib/internal/modules/esm/loader.js

```diff
@@ -584,21 +601,24 @@ class ESMLoader {
         }
       }
 
-      validateObject(ctx, `${hookErrIdentifier} context`);
+      if (ctx) validateObject(ctx, `${hookErrIdentifier} context`);
+    };
+    const validateOutput = (hookErrIdentifier, output) => {
+      if (typeof output !== 'object' || output === null) { // [2]
+        throw new ERR_INVALID_RETURN_VALUE(
+          'an object',
+          hookErrIdentifier,
+          output === null ? nullTypeForErr : output,
```

Comment:
I think a better way of handling that would be to modify `ERR_INVALID_RETURN_VALUE` implementation to take `null` into account.

Reply:
🤔 I think that should be a separate PR (to land before this one, and then consume here)